hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,164 @@
+---
+title: "Keywords as Identifiers"
+weight: 300
+---
+
+Langium treats keywords like `var`, `const` or `function` as keyword tokens, even at unintended locations according to your grammar.
+Additionally, these strings are always highlighted as _keyword_ (in blue in this guide, while other text is white) whenever they are used.
+Summarizing, these keywords cannot be used as values for names, identifiers or other properties by default and need to be explicitly enabled.
+This guide explains how to do that.
+
+Let's look at the "hello-world" example in the [playground](https://langium.org/playground?grammar=OYJwhgthYgBAEgUwDbIPYHU0mQEwFD6IB2ALiAJ6wCyauKAXPrC7ABQAOiIAzmsTwDUAXgAK3PsVgAfWKESJSAS2LAhwgOIgFy1QEoAVAG5C43vyatYAci7ni12MUiJhASQAiJ-Fp0rglqzWSKhojnaSwgDaZpIMngC6NgCE1t4AFkq49FKk3BAqYMiwGADKDLAA9AA6QpUmeSAFzsWeFZVRAPpgALQAXgCCPQBaCVHVAO6dCQb1hJnZJLCNzUU0ADKdAMIA8tTUAKIAcgAq7dU1BuM81aUzAPzVBhdzCznL%2BYXFpZu7%2B8dnKoXC5RAB61WI1RAMzmQA&content=A4UwTgzg9gdgBAKQIYxAKABIgDbaolEAQjTVElji1yjSA) or as a new local project created with `yo langium` (for details, how to set up your first Langium project, read [getting started](/docs/getting-started/)):
+
+![screenshot with the editor, an example and an error message](problem.png)
+
+Here, it is not possible to introduce a person whose name is "Hello", since `Hello` is a dedicated keyword of the language. Additionally, we cannot greet a person called "Hello" as well.
+The same applies to the keyword "person", but let's focus on enabling "Hello" as name for persons.
+
+To enable keywords as identifiers, you need to apply the following three steps:
+
+
+{{< toc format=html >}}
+
+
+## Step 1: Modify the grammar to explicitly parse keywords as property values
+
+The __first step__ is to modify the grammar to explicitly parse keywords as property values:
+At the moment, the parser rule for introducing persons looks like this:
+
+```langium
+Person: 'person' name=ID;
+terminal ID: /[_a-zA-Z][\w_]*/;
+```
+
+Note that the terminal rule for `ID` already covers the string "Hello",
+but since the parser rule for greeting persons uses "Hello" as keyword, the keyword takes precedence:
+
+```langium
+Greeting: 'Hello' person=[Person:ID] '!';
+```
+
+Roughly summarized, the background for this behaviour is, that Langium's internally used LL(k) parser implementation named [Chevrotain](https://chevrotain.io) first does _lexing_/tokenizing, i.e. splitting text into single tokens, e.g. words separated by white space.
+The actual _parsing_, i.e. the application of the parser rules, is performed afterwards on these tokens.
+Chevrotain uses regular expressions (regex) for splitting text into tokens.
+Since keywords are implemented as regex as well and take precedence, _all_ occurrences of "Hello" are treated as keywords for the parser rule for greetings,
```

Comment:
```suggestion
Since keywords are implemented as regex as well and take precedence, _all_ occurrences of "Hello" are treated as keywords for the parser rule named `Greeting`,
```

Reply:
changed